src/realm.js

```diff
@@ -436,7 +436,13 @@ export class Realm {
       this.timeoutCounter = this.timeoutCounterThreshold;
       let total = Date.now() - this.start;
       if (total > timeout) {
-        throw new FatalError("Timed out");
+        let error = new CompilerDiagnostic(
+          `total time has exceeded the timeout time: ${timeout}`,
+          this.currentLocation,
+          "PP0036",
+          "RecoverableError"
+        );
+        if (this.handleError(error) === "Recover") throw new FatalError();
```

Comment:
The reason I set as `===` is to make sure to throw since the original code just threw. If I set it as !==, it won't hit the throw. When I run with the timeout flag and `!==`, I get the following trace. I have to take another look if we shouldn't just throw in the timeout check
 
```at Realm.testTimeout (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/realm.js:302:23)
    at LexicalEnvironment.evaluateAbstract (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1285:16)
    at LexicalEnvironment.evaluate (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1278:20)
    at ForBodyEvaluation (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/evaluators/ForStatement.js:179:25)
    at exports.default (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/evaluators/ForStatement.js:99:12)
    at LexicalEnvironment.evaluateAbstract (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1290:20)
    at LexicalEnvironment.evaluate (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1278:20)
    at exports.default (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/evaluators/WhileStatement.js:8:15)
    at LexicalEnvironment.evaluateAbstract (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1290:20)
    at LexicalEnvironment.evaluate (/src/devenv/node-v8.11.1-darwin-x64/lib/node_modules/prepack/lib/environment.js:1278:20)
```

Reply:
Right... So then it shouldn't be a "RecoverableError" to begin with, but a "FatalError". And then the return value of handleError doesn't matter, and you should always throw.